src/apps/api/tests/test_analytics.py

```diff
@@ -0,0 +1,58 @@
+from django.urls import reverse
+from rest_framework.test import APITestCase
+
+from factories import SubmissionFactory, UserFactory, CompetitionFactory
+
+import datetime
+import json
+import pytz
+
+
+class AnalyticsTests(APITestCase):
+    def setUp(self):
+        self.user = UserFactory(username='user')
+        self.admin = UserFactory(username='admin', super_user=True)
+        self.day_range = 10
+        self.first_day = datetime.datetime(2019, 1, 1, tzinfo=pytz.UTC)
+
+    def create_datetime_with_days_offset(self, offset):
+        return self.first_day + datetime.timedelta(days=offset)
+
+    def date_range_tester(self, factory, date_field_name, count_field_name, **kwargs):
+        object_quantity = self.day_range
+        self.test_objects = []
+
+        for i in range(object_quantity):
+            datefield = {
+                date_field_name: self.create_datetime_with_days_offset(i),
+            }
+            self.test_objects.append(factory(**kwargs, **datefield))
+
+        query_params = [
+            self.create_datetime_with_days_offset(1).isoformat()[0:10],
+            self.create_datetime_with_days_offset(self.day_range - 2).isoformat()[0:10],
```

Comment:
`self.create_stuff().date().isoformat()`
then you don't need to do the slicing at the end

Reply:
Done.